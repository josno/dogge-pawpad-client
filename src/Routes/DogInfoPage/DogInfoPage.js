import React, { useState, useEffect } from "react";
import { IoIosArrowBack } from "react-icons/io";
import "react-responsive-modal/styles.css";

import { Link } from "react-router-dom";
import styled from "styled-components";

import DogsApiService from "../../services/api-service";

import ProfileSection from "../../Components/Sections/ProfileSection";
import MedicalSection from "../../Components/Sections/MedicalSection";
import NotesSection from "../../Components/Sections/NotesSection";
import AdoptionSection from "../../Components/Sections/AdoptionSection";
import FosterSection from "../../Components/Sections/FosterSection";

const DogInfoPage = (props) => {
	const dogId = props.match.params.dogId;
	const [status, setStatus] = useState("");
	const [update, setUpdate] = useState(false);

	useEffect(() => {
		async function getStatus() {
			const res = await DogsApiService.getDogStatus(dogId);

			setStatus(res.dog_status);
		}
		setUpdate(false);
		getStatus();
	}, [dogId, update]);

	const updateStatus = async () => {
		const res = await DogsApiService.getDogStatus(dogId);
		setStatus(res.dog_status);
	};

	return (
		<DogInfoPageStyles>
			<div className="buttons-section">
				<div className="back-button">
					<Link className="back-button-link" to="/dogs-list">
						<IoIosArrowBack /> Back
					</Link>
				</div>

				{/* <button className="delete-dog">Delete</button> */}
			</div>

			<div className="details-section">
				<div className="dog-details">
					<ProfileSection
						dogId={dogId}
						history={props.history}
						buttonStatus={(s) => setStatus(s)}
						setUpdate={setUpdate}
						update={update}
					/>
				</div>
				<div className="medical">
					<MedicalSection dogId={dogId} />
				</div>
				<div className="notes">
					<NotesSection dogId={dogId} update={update} setUpdate={setUpdate} />
				</div>
				<div className="adoption">
					<AdoptionSection
						dogId={dogId}
						status={status}
						updateStatus={() => updateStatus()}
						update={update}
						setUpdate={setUpdate}
					/>
				</div>
				<div className="foster">
					<FosterSection dogId={dogId} update={update} setUpdate={setUpdate} />
				</div>
			</div>
		</DogInfoPageStyles>
	);
};

const DogInfoPageStyles = styled.div`
	padding: 1rem;
	padding-top: 60px;
	height: 100%;
	max-width: 500px;
	margin: 0 auto;
	.dog-details,
	.medical,
	.adoption,
	.foster,
	.notes {
		position: relative;
		border: 1px solid black;
		margin: 10px 0px;
	}

	.buttons-section {
		display: flex;
		align-items: center;
	}
	.back-button {
		flex-grow: 1;
		.back-button-link {
			text-align: left;
		}
	}

	.archive,
	.delete-dog {
		margin: 0.8rem;
		position: relative;
		:hover {
			cursor: pointer;
		}

		background-color: #1f8392;
		color: white;
		border: 1px solid black;
		box-shadow: 0px;
		padding: 5px;
		font-weight: bold;
	}

	.details-section {
		display: flex;
		flex-direction: column;
	}

	@media (min-width: 1000px) {
		padding: 3rem;
		max-width: 1500px;

		.dog-details,
		.medical,
		.notes,
		.adoption,
		.foster {
			margin: 5px;
		}

		.details-section {
			display: grid;
			max-height: 800px;
			grid-template-columns: 0.8fr 1.1fr 1.1fr;
			grid-template-rows: 1.4fr 0.6fr;

			grid-template-areas:
				"dog-details medical notes"
				"dog-details adoption foster";

			.dog-details {
				grid-area: dog-details;
				display: flex;
				justify-content: center;
			}
			.medical {
				grid-area: medical;
				display: flex;
				flex-direction: column;
				position: relative;
			}
			.notes {
				grid-area: notes;
			}
			.adoption {
				grid-area: adoption;
			}
			.foster {
				grid-area: foster;
			}
		}
		.buttons-section {
			grid-area: buttons-section;
			display: flex;
		}
	}
`;

export default DogInfoPage;
