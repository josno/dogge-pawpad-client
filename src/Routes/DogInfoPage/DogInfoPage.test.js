import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import DogInfoPage from "./DogInfoPage";

describe("<AddNewDog />", () => {
	it("renders without crashing", () => {
		const div = document.createElement("div");
		ReactDOM.render(
			<BrowserRouter>
				<DogInfoPage
					match={{ params: { dogId: 1 }, isExact: true, path: "", url: "" }}
				/>
			</BrowserRouter>,
			div
		);
		ReactDOM.unmountComponentAtNode(div);
	});

	it("renders as expected", () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<DogInfoPage
						match={{ params: { dogId: 1 }, isExact: true, path: "", url: "" }}
					/>
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
