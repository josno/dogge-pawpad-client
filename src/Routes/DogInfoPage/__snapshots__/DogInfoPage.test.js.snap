// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddNewDog /> renders as expected 1`] = `
<div
  className="sc-fzoyAV iTNwVp"
>
  <div
    className="buttons-section"
  >
    <div
      className="back-button"
    >
      <a
        className="back-button-link"
        href="/dogs-list"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M217.9 256L345 129c9.4-9.4 9.4-24.6 0-33.9-9.4-9.4-24.6-9.3-34 0L167 239c-9.1 9.1-9.3 23.7-.7 33.1L310.9 417c4.7 4.7 10.9 7 17 7s12.3-2.3 17-7c9.4-9.4 9.4-24.6 0-33.9L217.9 256z"
          />
        </svg>
         Back
      </a>
    </div>
  </div>
  <div
    className="details-section"
  >
    <div
      className="dog-details"
    >
      <div
        className="sc-AxiKw bZsfrn"
      >
        <div
          className="img-container"
        >
          <img
            alt="dog"
          />
        </div>
        <svg
          className="icon edit-img-icon"
          fill="currentColor"
          height="1em"
          onClick={[Function]}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14,4 L20,10 L14,4 Z M22.2942268,5.29422684 C22.6840146,5.68401459 22.6812861,6.3187139 22.2864907,6.71350932 L9,20 L2,22 L4,15 L17.2864907,1.71350932 C17.680551,1.319449 18.3127724,1.31277239 18.7057732,1.70577316 L22.2942268,5.29422684 Z M3,19 L5,21 M7,17 L15,9"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
        <h1
          className="dog-name name-style"
        />
        <div>
          <svg
            className="icon edit-button"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,4 L20,10 L14,4 Z M22.2942268,5.29422684 C22.6840146,5.68401459 22.6812861,6.3187139 22.2864907,6.71350932 L9,20 L2,22 L4,15 L17.2864907,1.71350932 C17.680551,1.319449 18.3127724,1.31277239 18.7057732,1.70577316 L22.2942268,5.29422684 Z M3,19 L5,21 M7,17 L15,9"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
          <ul
            className="profile-details"
          >
            <li
              className="profile-list-item"
            >
              <span
                className="title"
              >
                Status:
              </span>
               
              <span
                className="value"
              />
            </li>
            <li
              className="profile-list-item"
            >
              <span
                className="title"
              >
                Gender: 
              </span>
              <span
                className="value"
              />
            </li>
            <li
              className="profile-list-item"
            >
              <span
                className="title"
              >
                Birthdate:
              </span>
              <span
                className="value"
              >
                April 14, 2021
              </span>
            </li>
            <li
              className="profile-list-item"
            >
              <span
                className="title"
              >
                Arrival:
              </span>
              <span
                className="value"
              >
                April 14, 2021
              </span>
            </li>
            <li
              className="profile-list-item"
            >
              <span
                className="title"
              >
                Microchip Date:
              </span>
              <span
                className="value"
              >
                April 14, 2021
              </span>
            </li>
            <li
              className="profile-list-item"
            >
              <span
                className="title"
              >
                Microchip:
              </span>
              <span
                className="value"
              />
            </li>
            <li
              className="profile-list-item"
            >
              <span
                className="title"
              >
                Tag:
              </span>
              <span
                className="value"
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="medical"
    >
      <h3
        className="sc-AxmLO loTDxG"
      >
        Medical
      </h3>
      <ul
        className="sc-AxhCb eFyOq"
      />
      <div
        className="sc-fzozJi cfGnXn"
      >
        <div
          className="sc-AxjAm ecLiXZ"
        >
          <svg
            className="undefined icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14,4 L20,10 L14,4 Z M22.2942268,5.29422684 C22.6840146,5.68401459 22.6812861,6.3187139 22.2864907,6.71350932 L9,20 L2,22 L4,15 L17.2864907,1.71350932 C17.680551,1.319449 18.3127724,1.31277239 18.7057732,1.70577316 L22.2942268,5.29422684 Z M3,19 L5,21 M7,17 L15,9"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </div>
        <div
          className="sc-AxjAm ecLiXZ"
        >
          <svg
            className="undefined icon"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,18 L12,6 M6,12 L18,12"
              fill="none"
              stroke="#000"
              strokeWidth="2"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="notes"
    >
      <div
        className="sc-fzplWN kycYaO"
      >
        <h3
          className="notes-title"
        >
          Notes
        </h3>
        <ul
          className="notes-filter"
        >
          <li>
            <button
              className="additional-title"
              onClick={[Function]}
            >
              additional
            </button>
          </li>
          <li>
            <button
              className="adoption-title"
              onClick={[Function]}
            >
              adoption
            </button>
          </li>
          <li>
            <button
              className="medical-title"
              onClick={[Function]}
            >
              medical
            </button>
          </li>
          <li>
            <button
              className="archive-title"
              onClick={[Function]}
            >
              archive
            </button>
          </li>
          <li>
            <button
              className="foster-title"
              onClick={[Function]}
            >
              foster
            </button>
          </li>
          <li>
            <button
              className="all-title"
              onClick={[Function]}
            >
              all
            </button>
          </li>
        </ul>
        <div
          className="notes-container"
        >
          <div
            className="sc-fznyAO hKaXBs"
          >
            <div
              className="sc-AxjAm ecLiXZ"
            >
              <svg
                className="undefined icon"
                fill="currentColor"
                height="1em"
                onClick={[Function]}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12,22 C17.5228475,22 22,17.5228475 22,12 C22,6.4771525 17.5228475,2 12,2 C6.4771525,2 2,6.4771525 2,12 C2,17.5228475 6.4771525,22 12,22 Z M12,18 L12,6 M6,12 L18,12"
                  fill="none"
                  stroke="#000"
                  strokeWidth="2"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="adoption"
    >
      <div
        className="sc-fznZeY jZLVg"
      >
        <h3
          className="adoption-title"
        >
          Adoption
        </h3>
        
      </div>
    </div>
    <div
      className="foster"
    >
      <div
        className="sc-fzokOt jqBIvf"
      >
        <h3
          className="foster-title"
        >
          Foster
        </h3>
        <div
          className="sc-fzqBZW hraBuz"
        />
      </div>
    </div>
  </div>
</div>
`;
